plutus-contract/test/Spec/Contract/TxConstraints.hs

```diff
@@ -242,3 +266,64 @@ mustReferenceOutputTxV2ConTest = do
 get3 :: [a] -> (a, a, a)
 get3 (a:b:c:_) = (a, b, c)
 get3 _         = error "Spec.Contract.TxConstraints.get3: not enough inputs"
+
+mustSpendScriptOutputWithReferenceV2ConTest :: Contract () EmptySchema ContractError ()
+mustSpendScriptOutputWithReferenceV2ConTest = do
+
+    utxos <- ownUtxos
+    myAddr <- Con.ownAddress
+    let ((utxoRef, utxo), (utxoRefForBalance1, _), (utxoRefForBalance2, _)) = get3 $ Map.toList utxos
+        ValidatorHash vh = fromJust $ Addr.toValidatorHash mustReferenceOutputV2ValidatorAddress
+        lookups = TC.unspentOutputs utxos
+               <> TC.plutusV2OtherScript mustReferenceOutputV2Validator
+        tx = TC.mustPayToOtherScript (ValidatorHash vh) (Datum $ PlutusTx.toBuiltinData utxoRef) (Ada.adaValueOf 5)
+          <> TC.mustSpendPubKeyOutput utxoRefForBalance1
+          <> TC.mustPayToAddressWithReferenceScript
+                myAddr
+                (ScriptHash vh)
+                Nothing (Ada.adaValueOf 25)
+    mkTxConstraints @Void lookups tx >>= submitTxConfirmed
+
+    -- Trying to unlock the Ada in the script address
+    scriptUtxos <- utxosAt mustReferenceOutputV2ValidatorAddress
+    utxos' <- ownUtxos
+    let
+        scriptUtxo = fst . last . Map.toList $ scriptUtxos
+        refScriptUtxo = fst . head . filter (isJust . Tx._ciTxOutReferenceScript . snd) . Map.toList $ utxos'
+        lookups = TC.unspentOutputs (Map.singleton utxoRef utxo <> scriptUtxos <> utxos')
+        tx = TC.mustReferenceOutput utxoRef
+          <> TC.mustSpendScriptOutputWithReference scriptUtxo unitRedeemer refScriptUtxo
+          <> TC.mustSpendPubKeyOutput utxoRefForBalance2
+    mkTxConstraints @Any lookups tx >>= submitTxConfirmed
+
+mustSpendScriptOutputWithReferenceTxV2ConTest :: Contract () EmptySchema ContractError ()
+mustSpendScriptOutputWithReferenceTxV2ConTest = do
+    let mkTx lookups constraints = either (error . show) id $ Tx.Constraints.mkTx @Any def lookups constraints
+
+    utxos <- ownUtxos
+    myAddr <- Con.ownAddress
+    let ((utxoRef, utxo), (utxoRefForBalance1, _), (utxoRefForBalance2, _)) = get3 $ Map.toList utxos
+        ValidatorHash vh = fromJust $ Addr.toValidatorHash mustReferenceOutputV2ValidatorAddress
+        lookups = Tx.Constraints.unspentOutputs utxos
+               <> Tx.Constraints.plutusV2OtherScript mustReferenceOutputV2Validator
```

Comment:
do we need this lookup? I think the ValidatorHash passed to mustPayToOtherScript is enough

Reply:
Yes, this is for creating the utxo with the reference script.